rust/src/container.rs

```diff
@@ -223,6 +297,28 @@ fn gv_nevra_to_string(pkg: &glib::Variant) -> String {
     }
 }
 
+/// Read the user.component xattr from a file path, returning None if not present
+fn get_user_component_xattr(file: &ostree::RepoFile) -> std::io::Result<Option<String>> {
+    let xattrs = match file.xattrs(gio::Cancellable::NONE) {
+        Ok(xattrs) => xattrs,
+        Err(_) => return Ok(None), // No xattrs available
+    };
+
+    let n = xattrs.n_children();
+    for i in 0..n {
+        let child = xattrs.child_value(i);
+        let key = child.child_value(0);
+        let key = key.data_as_bytes();
+        if key == COMPONENT_XATTR {
+            let value = child.child_value(1);
+            let value = value.data_as_bytes();
+            let value_str = String::from_utf8_lossy(&value).to_string();
```

Comment:
I think lossy conversions are generally only appropriate when doing things like taking corrupted/binary data and displaying it to a human.

In this case I think we should just explicitly fail if the user attempts to use non-UTF8 data for a component name.